library/ssl_msg.c

```diff
@@ -3292,11 +3323,74 @@ int mbedtls_ssl_prepare_handshake_record(mbedtls_ssl_context *ssl)
         }
     } else
 #endif /* MBEDTLS_SSL_PROTO_DTLS */
-    /* With TLS we don't handle fragmentation (for now) */
+#if defined(MBEDTLS_SSL_MAX_FRAGMENT_LENGTH)
```

Comment:
I'm not sure re-assembly should depend on `MAX_FRAGMENT_LENGTh` being enabled: even if we didn't send the extension, the peer is always free to fragment messages on its own initiative.